src/shims/files.rs

```diff
@@ -99,25 +62,14 @@ pub trait FileDescription: std::fmt::Debug + Any {
         throw_unsup_format!("cannot close {}", self.name());
     }
 
-    fn flock<'tcx>(
-        &self,
-        _communicate_allowed: bool,
-        _op: FlockOp,
-    ) -> InterpResult<'tcx, io::Result<()>> {
-        throw_unsup_format!("cannot flock {}", self.name());
-    }
-
     fn is_tty(&self, _communicate_allowed: bool) -> bool {
         // Most FDs are not tty's and the consequence of a wrong `false` are minor,
         // so we use a default impl here.
         false
     }
 
-    /// Check the readiness of file description.
-    fn get_epoll_ready_events<'tcx>(
-        &self,
-    ) -> InterpResult<'tcx, crate::shims::unix::linux::epoll::EpollReadyEvents> {
-        throw_unsup_format!("{}: epoll does not support this file description", self.name());
+    fn as_unix<'tcx>(&self) -> InterpResult<'tcx, &dyn UnixFileDescription> {
+        throw_unsup_format!("Not a unix file descriptor: {}", self.name());
```

Comment:
Shouldn't this be a panic? It would be a Miri bug to have a non-Unix FD oif the current target is a Unix target.

Reply:
Yes, probably. There wasn't a reason behind the choice of `unsup`, I just didn't know what kind of error this should be.

Reply:
Okay, then please make the return type `&dyn UnixFileDescription`, and "wrong" FD types should panic.